doc/src/manual/performance-tips.md

```diff
@@ -481,6 +481,64 @@ c = (b + 1.0f0)::Complex{T}
 does not hinder performance (but does not help either) since the compiler can determine the type of `c`
 at the time `k` is compiled.
 
+### Be aware of when Julia avoids specializing
+
+As a heuristic, Julia avoids automatically specializing on argument type parameters in three
+specific cases: `Type`, `Function`, and `Vararg`. Julia will always specialize when the argument is
+used within the method, but not if the argument is just passed through to another function. This
+usually has no performance impact at runtime and
+[improves compiler performance](@ref compiler-efficiency-issues). If you find it does have a
+performance impact at runtime in your case, you can trigger specialization by adding a type
+parameter to the method declaration. Here are some examples:
+
+This will not specialize:
+
+```julia
+function f_type(t)  # or t::Type
+    x = ones(t, 10)
+    return sum(map(sin, x))
+end
+```
+
+but this will:
+
+```julia
+function g_type(t::Type{T}) where T
+    x = ones(T, 10)
+    return sum(map(sin, x))
+end
+```
+
+These will not specialize:
+
+```julia
+f_func(f, num) = ntuple(f, div(num, 2))
+g_func(g::Function, num) = ntuple(g, div(num, 2))
```

Comment:
for consistency with the examples above and below, shall we replace the second line here with a comment, and have just the "f" and "g" examples 
```
This will not specialize:

```julia
f_func(f, num) = ntuple(f, div(num, 2))  # or f::Function
``

but this will:

```julia
g_func(g::G, num) where {G} = ntuple(g, div(num, 2))
```

Reply:
I think just becuase the comment was in a funny place? 

Because is followed a different structure, tt took me a while to realise that there was nothing special about the Function case